### DISCUSSION TITLE
Am I the only one having issues with "login with device" since Bitwarden android app redesign?

### DISCUSSION THREAD

**Discussion by onjour** [dani-garcia/vaultwarden#5182] [Q&A] [answered]
Hi all,

Basically I do not receive push notifications anymore to "login with device" on my phone since I updated to the recent Bitwarden official android app (I think it was last week) — the one with the drastic UI changes. 

I get the requests on the "pending login requests" section though, and I can approve them, but that's far less convenient. 

Did you face the same issue? Is there a way to make it work as intended? 

I'm up to date on my vaultwarden and bitwarden app.

Thanks for your feedback

**Comment by BlackDex** [comment]
There seem to be some issues with Bitwarden push from self-hosted instances. It also seems to happen on the Bitwarden Self-Hosted instances, where pushes are send, and also received by the mobile device, but the client doesn't act upon it.

It's not something we can fix that easily, and we may need to do some more debugging on this part. But i didn't had time yet my self.

**Comment by onjour** [comment]
Thanks @BlackDex for your answer and everything you do in this beautiful project. It's a bit frustrating because I was able to make it work before and it worked every time but now it's totally broken

**Comment by thetechjester** [comment]
Same issue here. Went back to version 2024.10.0 (11270) of the Android app. I get the notification, but when I tap it for approval it disappears. If I go into Pending login requests, it there but when I select it to approve, I get a message "AuthRequest doesn't exist" and just an option of OK.
I've gone back to using Master Password to unlock for now.

**Reply by BlackDex** [reply, marked as answer]
It is already fixed in `testing`

**Reply by thetechjester** [reply]
Awesome @BlackDex ! Thank you!

**Reply by larena1** [reply]
In testing of Vaultwarden? I am on testing and push isn't working here anymore too. Entries in the Android app will only update when manually syncing the vault. Last time I tested this was months back still with the old Android app.

**Reply by thetechjester** [reply]
Yes. I am using vaultwarden/server:testing docker with old version 2024.10.0 (11270) of the Android app and it is working perfectly! Thanks again @BlackDex !!!
It took me "forever" to figure out how to get my container to use **testing** instead of **latest** (I'm a Docker newb hahaha) but it's working great.

**Reply by Gerardv514** [reply]
> It took me "forever" to figure out how to get my container to use **testing** instead of **latest**

How did you do it

**Reply by thetechjester** [reply]
Let me start by defining a few things from the start that must FULLY be understood.

1. I am a Docker NEWB!!! Period. Full stop. 
2. Some of the things I say may not be accurately described (correct lingo or terminology) or sound correct to those that actually know what they are doing.
3. Because of 1 and 2, listen to me AT YOUR OWN RISK. I accept no responsibility and you assume all responsibility.
4. When in doubt, refer to steps 1,2 and 3 again. Especially 1 and 3.

With that said, I'm using Portainer.
My Vaultwarden was initially setup as a container ONLY. I needed to get the container into docker compose YAML format.
Everything I read /tried regarding Portainer and redirecting or changing from **latest** image to **testing** image failed.
I searched the Google-sphere high and low and finally ran across this:
https://github.com/Red5d/docker-autocompose

Using that method, I was able to get the compose YAML from the original **latest** Vaultwarden container, stop then delete the old Vaultwarden container, generate a new container/stack to pull/use the **testing** image by modifying the following line in the YAML file.

Look for the line    **image: vaultwarden/server:latest**
Changed it to     **image: vaultwarden/server:testing**


After that, I started the Vaultwarden docker again. It pulled the **testing** image and I've been running like that ever since.
Everything is working again like it was before.
I'd love to be able to use the nice new shiny Android app. But honestly, I take actual functionality of a great product, that is provided by the hard work of dedicated people like @BlackDex to pretty and shiny any day of the week. 
I can't thank and praise them enough for all they to do give us this to use.

Hope you find this helpful and remember... refer to step 1.

Hahaha!
Cheers!

**Reply by Gerardv514** [reply]
Ah so you ended up just creating a new container, which I knew was one way already. When you earlier said you got your container to use testing rather than latest, I took it as the existing container with latest and somehow you were able to keep it but just switch to testing, without having to create a new container.

**Reply by thetechjester** [reply]
I guess the answer is yes. What I also learned is this setup is WAY better. It created a Compose Stack as well as a Container. Now all I have to do is stop the existing container, modify the YAML in the Stack, and start it again. Allows me to switch back and forth at will. I've already done this to every Container I originally had. It make quick "tweaks" like this a breeze.

Cheers!

**Reply by onjour** [reply]
I can confirm that using 
- _[latest](https://github.com/dani-garcia/vaultwarden/releases/tag/1.32.5)_  (as of now 1.32.5) vaultwarden
- Bitwarden Android app v 2024.10.0 (11270) installed from Aurora Store
makes push working as expected, especially "login with device" requests.

Thanks everyone!

**Reply by larena1** [reply]
I'm on 2024.11.6 from Play Store and have granted the app notification permission but I'm not getting any. @BlackDex is this working for you again now?

**Reply by larena1** [reply]
```
11-19 15:23:08.289 net=1: Received com.x8bit.bitwarden
11-19 15:23:08.409 net=1: Acked com.x8bit.bitwarden:0
11-19 15:23:08.413 net=1: Successful broadcast to com.x8bit.bitwarden (id= time=63ms priority=NORMAL
```

My device receives the messages but the app doesn't post the notification

**Reply by onjour** [reply]
@larena1 did you also try with app version 11270 by any chance?

**Reply by larena1** [reply]
No. Is there any particular reason you use an old version?

**Reply by onjour** [reply]
@larena1 yes, @thetechjester and myself have had better results with this one.

**Reply by BlackDex** [reply]
I have tried debugging it a while back. Non Native clients work just fine, but the native clients seem to be broken for some reason. I have to try and dive a bit deeper into this to try and figure out why this happens. I have seen this also happening on the official Bitwarden Self-Hosted instances, so it looks like it isn't something specific to Vaultwarden it self, but more a Self-Hosted issue. For me the same, i see the FCM message being received, but the client doesn't seem to do do anything with it.

**Reply by larena1** [reply]
That's exactly what I found. They seem to discard the message for some reason. Official hosted servers are just fine with the new clients. 

https://github.com/bitwarden/android/issues/4319

Even filed an issue there but with limited success

**Reply by onjour** [reply]
> Non Native clients work just fine

What clients are you referring to exactly @BlackDex ? It's strange because I get my notifications just fine now 😕

**Reply by larena1** [reply]
@onjour I think that's because you're still using the old client and not the new one @BlackDex and I are discussing. First version is 2024.10.1

https://github.com/bitwarden/android/releases/tag/v2024.10.1

**Reply by onjour** [reply]
That's what I think as well @larena1, that's why I was suggesting you to use that version instead of the latest.

**Reply by larena1** [reply]
Yeah but the new one is much snappier and the old versions are EOL and won't get any more updates

**Reply by onjour** [reply]
As you wish @larena1, I prefer the ease of having push working rather than having a speed improvement 👍

**Comment by larena1** [comment]
@BlackDex the payload the app received is different

Official hosted: `{payload={"Id":"XXX","UserId":"XXXX","OrganizationId":null,"CollectionIds":[],"RevisionDate":"2024-11-20T14:02:50.089326Z"}, type=0}}`
Vaultwarden: `{payload={"id":"XXX","organizationId":null,"revisionDate":"2024-11-20T14:03:12.462690484","userId":"XXX"}, type=0}}`

Maybe that is why the app doesn't handle it?

**Reply by larena1** [reply]
https://github.com/bitwarden/android/blob/5ea17700b3d05fd28717df5f2359f9506b1c37f1/app/src/main/java/com/x8bit/bitwarden/data/platform/manager/model/NotificationPayload.kt#L26

Looks pretty much like it

**Reply by BlackDex** [reply]
Ah, they seem to still use PascalCase there?
And, maybe the date/time there is also incorrect!

Thanks, that helps me probably

**Reply by larena1** [reply]
Yeah should be an easy fix. Glad I could help :)

**Reply by BlackDex** [reply]
I haven't looked at the specific output generated yet, so this is nice! Thanks.

**Reply by BlackDex** [reply]
Thanks @larena1 i have fixed it via #5214

**Reply by Gerardv514** [reply]
> @BlackDex the payload the app received is different
> 
> Official hosted: `{payload={"Id":"XXX","UserId":"XXXX","OrganizationId":null,"CollectionIds":[],"RevisionDate":"2024-11-20T14:02:50.089326Z"}, type=0}}` Vaultwarden: `{payload={"id":"XXX","organizationId":null,"revisionDate":"2024-11-20T14:03:12.462690484","userId":"XXX"}, type=0}}`
> 
> Maybe that is why the app doesn't handle it?

Nice job

**Reply by thetechjester** [reply]
That's great! When will it be rolled out for testing?

**Reply by BlackDex** [reply]
Yes, in the `testing` tagged images.
